packages/jest-matchers/src/matchers.js

```diff
@@ -91,6 +91,32 @@ const matchers: MatchersObject = {
     return {message, pass};
   },
 
+  toChange(received: Function, expected: Function) {
```

Comment:
If this change gets accepted by the core team, I think you might want to put some protection around these variables to ensure they are functions, and throw meaningful errors if they aren't.

It's pretty typical with approaches like this to see somebody accidentally write this:

```
expect(myFunction()).toChange(array.length)
```

In that, I accidentally called the function instead of passing it to the `expect`